server/src/infra/repositories/job.repository.ts

```diff
@@ -116,12 +116,31 @@ export class JobRepository implements IJobRepository {
     ) as unknown as Promise<JobCounts>;
   }
 
-  async queue(item: JobItem): Promise<void> {
-    const jobName = item.name;
-    const jobData = (item as { data?: any })?.data || {};
-    const jobOptions = this.getJobOptions(item) || undefined;
+  async queueAll(items: JobItem[]): Promise<void> {
+    if (!items.length) {
+      return;
+    }
 
-    await this.getQueue(JOBS_TO_QUEUE[jobName]).add(jobName, jobData, jobOptions);
+    const itemsByQueue = items.reduce<Record<string, JobItem[]>>((acc, item) => {
+      const queueName = JOBS_TO_QUEUE[item.name];
+      acc[queueName] = acc[queueName] || [];
+      acc[queueName].push(item);
+      return acc;
+    }, {});
+
+    for (const [queueName, items] of Object.entries(itemsByQueue)) {
+      const queue = this.getQueue(queueName as QueueName);
+      const jobs = items.map((item) => ({
+        name: item.name,
+        data: (item as { data?: any })?.data || {},
```

Comment:
What's the reason you don't just do `data: item.data ?? {}`?

Reply:
I think this was just existing code and I wrote || over ?? Since I usually prefer it.

Reply:
Even then, why would you do `(item as { data: any }).data`?

Reply:
Not all items have a data property.

Reply:
Every job item has (at least an optional) data property, yes?

Reply:
I think that may be right, but this is existing code, which doesn't need to be modified as part of this pull request.